libs/bluegl/CMakeLists.txt

```diff
@@ -17,6 +17,28 @@ set(SRCS
     src/BlueGL.cpp
 )
 
+# detect aarch64 (ref: https://stackoverflow.com/a/12024211/1255535)
+set(DETECT_ARCH_SRC "
+#if defined(__ARMv8_A__) || defined(__ARM64_ARCH_8__) || \
```

Comment:
Could we use `uname` instead?

Also at the very least this should be done only on Linux, not Windows (we'll need this for macOS soon unfortunately).

Reply:
Done. Changed to `uname` and moved this check to the top-level cmake. 

`uname` is much cleaner than the previous approach. But one potential issue with `uname` is that it detects system arch, not the compiler target arch. So it might limit some cross-compilation use cases.